### PR TITLE
Remove deprecated APIs: InputFile.path()

### LINKED ISSUE #3478: Remove deprecated APIs: org.sonar.api.batch.fs.InputFile.path()
Remove `org.sonar.api.batch.fs.InputFile.path()` from EncodingPerFileTest, GeneratedFileFilter, GeneratedFileFilterTest, ProtobufImporter

>@deprecated since 6.6 use #inputStream() for file content, #filename() for file name, #uri() for an unique identifier, and #toString() for logging

To really deprecate `.path()` concept to unique identifiers concept, the class `FileMetadataImporter` and it's dependencies (`AbstractGlobalProtobufFileProcessor`) have to be reworked.

### LINKED ISSUE #3479: Remove deprecated APIs: Remove `org.sonar.api.batch.fs.InputFile.{relativePath(), file()}` [TEST only]
`org.sonar.api.batch.fs.InputFile.relativePath()` from ProtobufDataImporterTest
`org.sonar.api.batch.fs.InputFile.file()` from SensorContextUtilsTest

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #3478
Fixes #3479

### CODE REVIEW

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilterTest.java [hunk 1]
```diff
@@ -58,13 +58,14 @@ public void setUp() {
       SonarRuntimeImpl.forSonarQube(Version.create(7, 9), SonarQubeSide.SERVER, SonarEdition.COMMUNITY)) {
     };
     MapSettings settings = new MapSettings(new PropertyDefinitions(definitions.create()));
-    defaultConfiguration = new AbstractSolutionConfiguration(settings.asConfig(), "cs") { };
+    defaultConfiguration = new AbstractSolutionConfiguration(settings.asConfig(), "cs") {
```
Comment (andrei-epure-sonarsource) [inline sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilterTest.java:61]: is this change necessary?

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporterTest.java [hunk 2]
```diff
@@ -68,26 +69,32 @@ public void prepare() throws Exception {
...
       .build();
-    tester.fileSystem().add(inputFile);
+    tester.fileSystem().add(firstFile);
 
-    when(fileLinesContextFactory.createFor(inputFile)).thenReturn(fileLinesContext);
+    secondFile = new TestInputFileBuilder(tester.module().key(), "Data.cs")
+      .setLanguage("cs")
+      .initMetadata(new String(Files.readAllBytes(csFile), StandardCharsets.UTF_8))
+      .build();
+    tester.fileSystem().add(firstFile);
```
Comment (andrei-epure-sonarsource) [inline sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporterTest.java:82]:
```suggestion
    tester.fileSystem().add(secondFile);
```

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporterTest.java [hunk 3]
```diff
@@ -68,26 +69,32 @@ public void prepare() throws Exception {
...
+
+    when(fileLinesContextFactory.createFor(firstFile)).thenReturn(fileLinesContext);
```
Comment (andrei-epure-sonarsource) [inline sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporterTest.java:84]: shouldn't you do lines 84 and 88 for `secondFile` as well?

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilterTest.java (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/GeneratedFileFilterTest.java:61]: Functionally no. Auto-Formatting did this. I believe I've set it as it should be according to [sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style-configuration-for-intellij)

I'm not sure what settings does that, but [this line](https://github.com/SonarSource/sonar-developer-toolset/blob/7ef94ba29b0b47d1ccfd072bf926f75942b37fa1/eclipse/sonar-formatter.xml#L119) seems to be the one explicitly requiring this (since the `{ }` is an anonymous type declaration).

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporterTest.java (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporterTest.java:82]: `secondFile` was intended here, it was a post-copy/paste issue.

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporterTest.java (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporterTest.java:84]: Line 88: It's good idea to move the scaffolding here to simplify the test below. No harm done to other tests.

Line 84: This is now needed after new line 90 was added.

**Comment by pavel-mikula-sonarsource** [comment]
Hmm, I wonder why SonarLint 4.10 (that's supposed to have AssertJ rules from v4.9) didn't show me these.

**Comment by pavel-mikula-sonarsource** [comment]
@andrei-epure-sonarsource I did nothing to modify the reported regression and the CI passes (just as it passed locally and in the previous commit in Azure). So it was 50% random infrastructure issue and 50% some nonreproducible nondeterminism in the code or IT 😕 We should keep eye on this particular error message.

**Comment by andrei-epure-sonarsource** [comment]
Basically we should keep an eye on the detection of auto-generated code. From time to time we have community reports about this.
